packages/evm/src/errors.ts

```diff
@@ -38,12 +41,38 @@ export enum ERROR {
   INVALID_PROOF = 'kzg proof invalid',
 }
 
-export class EvmError {
-  error: ERROR
-  errorType: string
+export enum EvmErrorCode {
+  UNSUPPORTED_FEATURE = 'EVM_ERROR_UNSUPPORTED_FEATURE',
+  RUNTIME_ERROR = 'EVM_ERROR_RUNTIME_ERROR',
+}
+
+type EvmRuntimeErrorType = {
+  code: EvmErrorCode.RUNTIME_ERROR
+  reason: RuntimeErrorMessage | EOFError
+} & (
+  | { reason: RuntimeErrorMessage.REVERT; revertBytes: Uint8Array }
+  | { reason: Exclude<RuntimeErrorMessage, RuntimeErrorMessage.REVERT> | EOFError }
+)
+
+export type EvmErrorType = { code: EvmErrorCode.UNSUPPORTED_FEATURE } | EvmRuntimeErrorType
 
-  constructor(error: ERROR) {
-    this.error = error
-    this.errorType = 'EvmError'
+export class EvmError extends EthereumJSError<EvmErrorType> {
```

Comment:
Side note, but I think we might want to migrate to the EVM (rather than Evm) capitalization as we've done for some others (Json -> JSON, etc.)